### ISSUE TITLE
Cart/Checkout: update feedback prompt and include CES

### ISSUE THREAD

**Issue by pmcpinto** [woocommerce/woocommerce#42091]
**Description:** We need to enhance our existing feedback loops to ensure we collect valuable feedback. The current prompt for collecting user feedback in the block editor requires improvement. The copy needs to be improved, and directing users to GitHub is a friction point, as merchants need to have GitHub account to share feedback.

![Captura de ecrã 2023-11-16, às 16 53 08](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/0f40f80d-bb2c-4021-a145-c9513f05c559/e84bc96c-0b94-44ef-93ac-1cee8c6837ed?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi8wZjQwZjgwZC1iYjJjLTQwMjEtYTE0NS1jOTUxM2YwNWM1NTkvZTg0YmM5NmMtMGI5NC00NGVmLTkzYWMtMWNlZThjNjgzN2VkIiwiaWF0IjoxNzg1MjU4NDA2LCJleHAiOjE4MTY4Mjg5NjZ9.6U24wtl_L1V_hLxs1E1RWfaHcT2FNqBI1G-2JFPVXEk)

**Acceptance criteria:**

- [ ] To remove friction, we'll use the [Customer Effort Score](<https://www.npmjs.com/package/@woocommerce/customer-effort-score>) instead of redirecting users to GitHub or the feature request page. We'll have access to a good mix of qual/quant data, and automatically feed the content to the FSOR. For example, here's what we display in the new product form:

![Captura de ecrã 2023-11-16, às 16 52 17](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/bd015bdf-ffe7-475b-95cc-8cf69040db3e/65f1dc69-c086-46ae-aba6-33d972e2256d?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi9iZDAxNWJkZi1mZmU3LTQ3NWItOTVjYy04Y2Y2OTA0MGRiM2UvNjVmMWRjNjktYzA4Ni00NmFlLWFiYTYtMzNkOTcyZTIyNTZkIiwiaWF0IjoxNzg1MjU4NDA2LCJleHAiOjE4MTY4Mjg5NjZ9.TA4DdlomowQQswxmfcaqZ5nnbJZt-4LsTo4vt2b1fHQ)

- [ ] We should use the following copy in the CES modal:<br>Header: What do you think of the cart/checkout/order confirmation\* editor?

1. The cart/checkout/order confirmation\* editing screen is easy to use
2. The cart/checkout/order confirmation\* editing screen's functionality meets my needs

\*- this part of the copy should change depending on the block edited by the user.

- [ ] The feedback prompt copy should also be updated: We're making improvements to checkout customization to provide merchants with the tools they need. Share your feedback.

**Comment by pmcpinto** [comment]
@elizaan36 you've mentioned that we should consider having the same feedback prompt across all the Woo pages. If we do that, I'm afraid we'll end up with generic feedback or suggestions completely unrelated to the checkout flows. For now, I'd prefer a prompt more focused on the checkout experience when the user interacts with one of the checkout-related blocks. Let me know what you think.

**Comment by elizaan36** [comment]
> you've mentioned that we should consider having the same feedback prompt across all the Woo pages. 

I think keeping the feedback prompt on the checkout flows makes perfect sense, I suggested updating the copy to something more easily understood.

> We're making improvements to store editing to provide merchants with the tools and customization options they need. _Share your feedback._

Or specific to checkout:
> We're making improvements to checkout customization to provide merchants with the tools they need. _Share your feedback._

I really like the idea of using the Customer effort score prompt in the editor. That seems easy and streamlined, from the merchant perspective and for our team to view/analyze the feedback in the FSOR. 💯

**Comment by pmcpinto** [comment]
> I really like the idea of using the Customer effort score prompt in the editor. That seems easy and streamlined, from the merchant perspective and for our team to view/analyze the feedback in the FSOR. 💯

I agree. @ralucaStan before we make a decision, it would be good to have a sense of the effort required to fire the Customer Effort Score rather than redirecting the user to an external link. It's available as an NPM package: https://www.npmjs.com/package/@woocommerce/customer-effort-score
Thanks!

**Comment by ralucaStan** [comment]
@pmcpinto the Customer Effort Score seems to be a [package](https://www.npmjs.com/package/@woocommerce/customer-effort-score) that we can import and use out of the box, as it gives us the component. This effort should be small.

Seeing the package that only mentioned offering the component for CES my 1st question was, ok, but where do we save the data? Digging deeper I saw that there is a [component](https://github.com/woocommerce/woocommerce/blob/2f7bd83ae229a999456e3e713af4d4a98e2bd28d/packages/js/customer-effort-score/src/components/customer-effort-score-tracks/index.js#L15) that allows sending the data to Tracks.

@pmcpinto did you work with this before? Is the expected result that the data gets saved in Tracks? In this case, we'd need to define some parameters.

**Comment by pmcpinto** [comment]
Yes. The data should be saved in Tracks. This is how it works in other flows where we trigger the CES.

We should also review the copy we display in the prompt and the modal. cc @elizaan36 in case you have some suggestions apart from what you shared in this comment: https://github.com/woocommerce/woocommerce/issues/42091.

**Comment by ralucaStan** [comment]
This should be shipped ideally in WooCommerce 8.5. In WooCommerce 8.4 we will be doing this https://github.com/woocommerce/woocommerce-blocks/issues/11990

**Comment by elizaan36** [comment]
> cc @elizaan36 in case you have some suggestions apart from what you shared in this comment

I'm not entirely sure of the best way to describe the cart and checkout blocks for the user. Some ideas for copy for the modal:

> Share feedback on the new checkout editor
> The checkout editor's functionality meets my needs

OR

> Share feedback on the new store editor
> The store editor's functionality meets my needs

OR

> Share feedback on checkout customization
> Woo's checkout customization features meet my needs.

**Comment by pmcpinto** [comment]
I think we should go with the first one. So:

Header: What do you think of the cart/checkout/order confirmation* editor?
1. The cart/checkout/order confirmation* editing screen is easy to use
2. The cart/checkout/order confirmation* editing screen's functionality meets my needs

*- this part of the copy should change depending on the block edited by the user.

I'll update the issue description.

**Comment by jarekmorawski** [comment]
There's a broader initiative to align all feedback forms and use a single Crowdsignal-based solution across admin, blocks, and other product areas. I worked on a design that could apply to cart and checkout.

Here's the Figma (qXJqJjr0LvCyNQkvr783nP-fi-BoeKvqCP7ko0qDoS-11) if you'd like to take a closer look and share feedback, and here's how it'd apply to cart and checkout.

<img width="853" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/03607fd1-72ca-4f86-a4ca-b1a44cfe2c5a">

The tl;dr is that we'd have a new feedback link in the block details section (for cart and checkout, I thought it'd make sense to show it at two levels: the actual block and the overarching parent block). When clicked, we'd open a modal with a product form (it's based on UMUX Lite, but shorter; context: p1717569393454749-slack-C05UMTA7W0P).

**Comment by ralucaStan** [comment]
Thank you @jarekmorawski. Does this proposal have a green light already?

**Comment by pmcpinto** [comment]
@elizaan36 @nikkivias in case you want to take a look at Jarek's proposal. Thanks

**Comment by pmcpinto** [comment]
> Thank you @jarekmorawski. Does this proposal have a green light already?

IMO, we can move forward with Jarek's proposed changes, as they align with what will be implemented for the other WooCommerce blocks, as discussed with @manospsyx 

@jarekmorawski @nikkivias @elizaan36 do you agree?

**Comment by elizaan36** [comment]
Really nice proposal, @jarekmorawski !

@pmcpinto Yes agreed, I think this could create nice consistency in the experience.

**Comment by ralucaStan** [comment]
@manospsyx is there any other team currently working/with plans to work on implementing the new layout?

**Comment by manospsyx** [comment]
@ralucaStan every Store Editing team is encouraged to allocate some capacity towards this starting with the team(s) working on the following blocks:

- Product Collection
- Product Filters
- Cart, Checkout and Mini Cart
- Key product template blocks (e.g. Product Price, Add to Cart with Options, Additional Information, Reviews, etc)

We've chatted about this a few times with @kmanijak @imanish003 and the folks working on the Product Collection and Product Filter blocks -- we don't need to schedule this with urgency; but ultimately, prioritizing this will enable us to prioritize better and understand if we're building the right things right :) 

Would be great to coordinate things with @sunyatasattva to ensure that our teams share code/knowledge as we add this to more blocks!

**Comment by pmcpinto** [comment]
@sunyatasattva has there been any development on the part of the Store Editing teams on this? The sooner this is implemented, the better. So we can improve the quality of our inputs.

**Comment by kmanijak** [comment]
> @sunyatasattva has there been any development on the part of the Store Editing teams on this?

I know @imanish003 started work on this. So he may be best suited to give some update. He posted some questions and concerns in pdnLyh-7gl-p2.

**Comment by imanish003** [comment]
Hey @pmcpinto,

I had initially aimed to have a PR ready for this during the last cooldown (pdnLyh-7nT-p2#comment-5440). Unfortunately, the Woo Meetup prevented me from making as much progress as I'd hoped.

As @kmanijak pointed out, we've addressed all the questions in this P2 (pdnLyh-7gl-p2), and there are no blockers. I plan to complete the implementation for Product Collection block in next cooldown. Once we've successfully implemented it, I'm confident we'll be able to apply the same approach to other blocks.

**Comment by pmcpinto** [comment]
Hey @imanish003 thanks for the update. Do you also plan to apply it to the cart and checkout blocks, or should Rubik handle it? cc @opr

**Comment by imanish003** [comment]
Hey @pmcpinto,

My focus is on implementing the feedback form for the Product Collection block. I believe this implementation will serve as a template that can be easily adapted for other blocks. I am open to implementing it for other blocks too. I think it's a good idea to complete the Product Collection block implementation first.

After successfully integrating the feedback form into the Product Collection block, I'll provide a more detailed assessment of what it would take to replicate this for other blocks. I'll keep you updated on the progress. Does this sound good to you?

**Comment by pmcpinto** [comment]
Hey @imanish003, do you have news on when you plan to integrate the modal in the first block?

**Comment by imanish003** [comment]
Hey @pmcpinto, 

I have integrated CES into the Product Collection block in PR #52874. The PR has already been approved and is ready to be merged. However, there are some failing tests, likely due to changes in the pnpm lock file. Since I'm currently on Support Rotation, I haven't had much time to investigate the issue. I will try to get it merged tomorrow. After that, I plan to publish a P2 about a reusable component I created and how it can be utilized in other blocks.

**Comment by pmcpinto** [comment]
Thanks for the update. cc @ralucaStan I'll add it to the cycle 10 scope for estimation.

**Comment by pierorocca** [comment]
Are there other styles available than the one used for the new product editor?There was specific negative feedback given to this specific survey during a recent demo. I agree with that demo feedback that the emojis don't look professional, and the 2nd question doesn't quite fit in a CES.  [Airtable](https://airtable.com/appUgxrLNUxZieiMh/pagF6eGmyOWTk8CjL/form) offers a decent rating option. Crowdsignal isn't suitable without custom work. 

[Userpilot offers some good examples](https://userpilot.com/blog/customer-effort-score-survey-template/) for inspiration that are clean, professional, and mobile friendly. This is likely a good issue to get @ifinoi's assistance on a visually harmonized approach for in-app surveys? 

![Image](https://github.com/user-attachments/assets/7b8ec095-e1e1-422a-91ce-30d5da7c6b25)
![Image](https://github.com/user-attachments/assets/0574eea5-86d6-454e-8d81-df9ba8cd765e)
![Image](https://github.com/user-attachments/assets/d42076f5-d2ae-4560-96dc-9c54a2f70da8)

**Comment by imanish003** [comment]
@pierorocca FWIW I previously discussed the emoji design in an internal RFC (pdnLyh-7gl-p2#comment-5388) where we decided to go with emojis. Since the design (used in product editor) is implemented in the `@woocommerce/customer-effort-score` package and used across WooCommerce, IMO any design changes should be made at the package level to maintain consistency.

Sidenote: I've recently [merged a PR](https://github.com/woocommerce/woocommerce/pull/52874) that integrates the feedback form into the Product Collection block using this package. As part of that work, I created a reusable `CesFeedbackButton` component that can be easily added to other blocks. I plan to publish a P2 post this week documenting how others can use this component. CC: @ralucaStan

**Comment by pierorocca** [comment]
Thanks @imanish003 might be good practice to expand future RFCs to a P2 with a broader scope should something be implemented as a Woo wide package. Otherwise the feedback scope will be too narrow IMO. Thanks for the context.

**Comment by keoshi** [comment]
Hey, all! Picking this one up from design and circling back to the open question around the visuals.

I have a preference to _not_ use emojis in these scenarios for a couple of reasons:

- Numbers are universal and provide a scale inherently. Understood immediately, no room for misinterpretations. A bit drier for sure, but also more professional, as it was mentioned above.
- Even though emojis are omnipresent and universal in their own right, each OS/device will use its own emojis and the final visual can differ wildly from user to user. I still think it takes a second look to interpret the scale with representations of emotions, but I might be in the minority (showing my age).
- Emojis readability depend on their size, while numbers have a lot more flexibility (and legibility) built into them.

Emojis | Numbers
--|--
<img width="1400" height="1464" alt="Image" src="https://github.com/user-attachments/assets/4fff7a1a-da18-440c-a88c-4afb409bb8fa" /> | <img width="1400" height="1464" alt="Image" src="https://github.com/user-attachments/assets/3e4157c2-19cb-4c7e-aaeb-f2caeb8f22bf" />

With that, I would also like to make a recommendation around isolating each option within its own button. Well defined boundaries using a background color, space between them to create extra separation. We could even use the brand color for the selected state for a bit of personality. Otherwise, a neutral color would work as well.

Neutral | Purple
--|--
<img width="1372" height="608" alt="Image" src="https://github.com/user-attachments/assets/c5246cd3-dce2-455a-86ea-1069650e9ea1" /> | <img width="1372" height="608" alt="Image" src="https://github.com/user-attachments/assets/2d66e000-9c4a-4770-bb19-aa11643ebed7" />

WDYT?

**Comment by elizaan36** [comment]
I like it @keoshi! One question is around where this feedback would go now, since it's been a while since the original issue. Maybe something to discuss with @pierorocca and the dev team.

I'd also consider a quick audit of other areas in the Woo admin where we're using emoji scales and try to make a more holistic shift to the number scale. The Customer effort score is a defined package that was created a few years ago, so we could update the design at the package level to be consistent in our feedback loop systems everywhere.
